### DISCUSSION TITLE
Measurable CSSDB Stages

### DISCUSSION THREAD

**Discussion by jonathantneal** [csstools/postcss-plugins#194] [Ideas]
How would we feel about agreeing to a measurable way to qualify the "stage" of a feature in the CSSDB?

## Stage 0: Aspirational

I _think_ this stage is well defined. Here is how I would qualify it:

- The feature is defined in a **Unofficial Draft** or **Editor’s Draft** specification.
- The specification exists on a public URL.
- The specification authorship belong to a **W3C Working Group Member**.
- The source code for the draft can be traced to a commit by that **W3C Working Group Member**.

## Stage 1: Experimental

I _think_ this stage is well defined. Here is how I would qualify it:

- The feature is defined in a **Editor’s Draft** or **Working Draft** specification.
- The specification exists on a public URL whose domain is `csswg.org` or `w3.org`.

## Stage 2: Allowable

I _think_ this stage is **not well defined**. Here is how I would qualify it so that it is well defined:

- The feature is defined in a **Editor’s Draft** or **Working Draft** specification.
- The specification exists on a public URL whose domain is `csswg.org` or `w3.org`.
- A previous version of the specification (from the **Previous Versions** metadata) exists on a public URL whose domain is `w3.org`.
- The feature is defined in the previous version of the specification.

An example of this is [Custom Media Queries](https://cssdb.org/#custom-media-queries), which are listed as Stage 1, but would actually be Stage 2. <sup>[reference](https://www.w3.org/TR/2021/WD-mediaqueries-5-20211218/)</sup>

## Stage 3: Embraced

I _think_ this stage is well defined. Here is how I would qualify it:

- The feature is defined in a **Candidate Recommendation** specification.
- The specification exists on a public URL whose domain is `w3.org`.
- The feature is implemented or intended to be implemented by at least 2 recognized browser vendors, where implementation is known from MDN data, or where "intent to implement" is communicated in either a recognized "Platform Status" domain or a public GitHub Issue/PR URL (_if someone really catches it fast_).

## Stage 4: Standardized

I _think_ this stage is well defined. Here is how I would qualify it:

- The feature is defined in a **Recommendation** specification.
- The specification exists on a public URL whose domain is `w3.org`.
- The feature is implemented by all recognized browser vendors, where implementation is known from MDN data.

## Rejected

I _think_ this non-stage is well defined, and I would qualify it with the exact wording it already has:

- Any specification that has been rejected or neglected by its editor, or formally rejected by a **W3C Working Group**.

**Comment by Antonio-Laguna** [comment]
@jonathantneal this is great and gave me great insight. I've created a PR for CSSDB here: https://github.com/csstools/cssdb/pull/78

**Comment by romainmenke** [comment]
We now also have [`minimumVendorImplementations `](https://github.com/csstools/postcss-plugins/tree/main/plugin-packs/postcss-preset-env#minimumvendorimplementations) in `postcss-preset-env`.

This is measurable and updates automatically based on data from mdn/caniuse.

----------

We haven't updated the `stage` data in a long time 🤔
